### DISCUSSION TITLE
Please re-add the option to use classes / instances (old method)

### DISCUSSION THREAD

**Discussion by hypervoxel** [TheLastBen/fast-stable-diffusion#490] [General]
Please re-add the option to use the previous method of classes / instances (old method). I got better results that this session / instances method

**Comment by ghost** [comment]
I agree with this. I've run some experiments myself comparing the old and new method, followed the instructions perfectly. Used the same dataset, same number of steps, same model. Labeled my files correctly, with the token and a number, (token (1), token (2) and so on. Token was an entirely unknown, password like string. I'm specifying this because creator likes to blame bad results on following the instructions incorrectly. 

New method has a more vague understanding of the character, old method can actually copy some ones likeness very closely. A lot of TheLastBen's examples were ran on characters or actors that Stable Diffusion already had prior knowledge of, aiding in the training process. To truly test it, he should train on an entirely unknown character, a pop star of the 2020s perhaps. And train with the old method and the new one, and make the both the models public. Then, maybe the creator and everyone else will finally see, the new method is worse.

Please, if anyone can back us up here! The more people agreeing, more likelihood we can get the old method back.

I'd also like if he could implement the option to use the old 1.4 model.

To TheLastBen, please don't take this personally or as an attack as you tended to do in the reddit thread. It's just constructive criticism. As a developer, you should be able to handle that without shooting down everyone who tries to help the project.

**Reply by TheLastBen** [reply]
> A lot of TheLastBen's examples were ran on characters or actors that Stable Diffusion already had prior knowledge of, aiding in the training process.

That's completely wrong, training with a different instance name makes it a completely new character for stable diffusion. I bet even if you train on a known actor, you won't succeed

**Reply by ghost** [reply]
> > A lot of TheLastBen's examples were ran on characters or actors that Stable Diffusion already had prior knowledge of, aiding in the training process.
> 
> That's completely wrong, training with a different instance name makes it a completely new character for stable diffusion. I bet even if you train on a known actor, you won't succeed

Can you identify the reason as to why I am not succeeding? Did you identify an issue within my workflow?

If it's wrong, you could easily prove it. Just train a model on a completely unknown concept, yourself or something like that. Some pop star or actor who became known well after the dataset used for SD was created. 

I trained models on myself, and several actors/actresses. I never once got good results with the new method. The character was always vauge and distorted.

Just started a new old method model, trained for HALF the amount of steps. 80% text encoder, class woman, 200 class images. Jaw dropping results. Looks like the person and the prompt dosen't require ridiculous amounts of engineering to get a good image.

**Comment by TheLastBen** [comment]
@hypervoxel 
https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/Dreambooth/fast_DreamBooth-Old-Method.ipynb

**Reply by ghost** [reply]
Well now I feel bad, I had no clue this existed. Thanks for making this.

maybe add that to the readme so ppl can find it easier

**Reply by hypervoxel** [reply]
Hello @TheLastBen , thank you for your amazing work !
I just tried the link to fast_DreamBooth-Old-Method, but unfortunately the link does not work
If this notebook exists, please post the link again. Thank you !

**Reply by TheLastBen** [reply]
Let's try the new method, I'll help you through it, if you don't get good results, I'll bring back the old one.

**Reply by hypervoxel** [reply]
I appreciate the help, but without sharing a lot of information I don't think it would be useful. The new method, when using the recommended training steps is x5 slower for me. Also, no longer using the class method, makes much more limited instance results. If need be I'll just use much older versions of your notebook.

**Reply by TheLastBen** [reply]
how come x5 slower ?

**Reply by SAC020** [reply]
I would like to share my own experience on this, and maybe get some advice: I am training for a NSFW position using a sample set of around 110 pics of different women in that specific position. Using SD20 720px, new method. I am not using text encoding (disabled).

I am now at 9200 steps (80x) and I am getting:
- reasonably good position replication
- poor limbs (hands, feet) - somewhat better than vanilla SD, but still deformed
- good styling (backgrounds, clothing)
- **worst problem is that I get garbled faces, with or without face fix**

I am not sure if I should persist in additional training, or there's something fundamentally wrong with the approach? Is it mixing the various faces I am feeding it, or why are the faces garbled? 

Not sure if the old method was better suited? (as I didn't have too much time to test it). If I were to use the old method, should I train for character (as I have multiple characters) or style?

I'd be happy to share details (ckpt, image set) but "offline", as they are NSFW.

**Reply by TheLastBen** [reply]
SD 2.0 was filtered aggressively against NSFW material, so you won't get much results with NSFW training.

**Reply by SAC020** [reply]
> SD 2.0 was filtered aggressively against NSFW material, so you won't get much results with NSFW training.

I see your point, but I don't think the filtering affects faces, and, as mentioned, the pose is quite well-trained, it's faces and limbs I have isssue with

![image](https://user-images.githubusercontent.com/50420272/205650788-40356db1-f65a-49a5-a6d7-64c07535e50d.png)

![image](https://user-images.githubusercontent.com/50420272/205651099-186aae49-1bd5-433c-bd01-52dca7946739.png)

**Reply by TheLastBen** [reply]
yes the V2 model struggles a lot with human anatomy

**Reply by hypervoxel** [reply]
> how come x5 slower ?

@TheLastBen I think due to this recommendation ? "Total Steps = Number of Instance images * 200,"
This makes training take 6 hrs - 8hrs for me on a high memory notebook. Last method training took on 1-2hrs

**Reply by TheLastBen** [reply]
you don't have to do the whole steps at once, do 25%, test the model, if not enough resume the training and so on.

**Reply by hypervoxel** [reply]
@TheLastBen ok, will try that. But I'm also missing class/instance instead of just instance. I was able to produce more variety. This named instances method doesn't seem to product as much variety

**Reply by hypervoxel** [reply]
@TheLastBen if I've calculated 1000 steps, and I want to resume to do 1000 more, what do I input, 1000 or 2000 ?

**Reply by TheLastBen** [reply]
It will train for exactly the number of steps you input, you put 1000, it will train for a 1000 more, you put 5000 it will train for 5000 more steps ... etc

**Reply by TetraOrb** [reply]
> > SD 2.0 was filtered aggressively against NSFW material, so you won't get much results with NSFW training.
> 
> I see your point, but I don't think the filtering affects faces, and, as mentioned, the pose is quite well-trained, it's faces and limbs I have isssue with
> 
> ![image](https://user-images.githubusercontent.com/50420272/205650788-40356db1-f65a-49a5-a6d7-64c07535e50d.png)
> 
> ![image](https://user-images.githubusercontent.com/50420272/205651099-186aae49-1bd5-433c-bd01-52dca7946739.png)

I used to train the 1.5 model on my own highly stylised images (wanted it to replicate my style) and I also have this face problem. I thought it was just due to the dataset itself. When I'm making specificaly a portrait it works, but the face is always samish... but every time it generates a full-half body character, the face is a mess...
I also still have no clue, how much text encoder training i should do if any...

![00112-3272044584-portrait of a happy lady art by tetraorb](https://user-images.githubusercontent.com/115459677/206731708-7de500e6-d02e-4a54-83a8-af093d1d3e6a.png)
![00118-3240992275-closeup portrait of a happy lady, beautiful face, violet eyes, blond hair, dramaric light art by tetraorb](https://user-images.githubusercontent.com/115459677/206731721-1c57204d-47b2-4520-bc66-c146b1d4ca0d.png)

**Reply by TheLastBen** [reply]
use the latest colab, I updated the method, for a style, set the text encoder to 350 steps, and the total steps to 2000-3000

**Reply by TetraOrb** [reply]
Thank you! I'll try it right away!

**Reply by SAC020** [reply]
> use the latest colab, I updated the method, for a style, set the text encoder to 350 steps, and the total steps to 2000-3000

No matter the number of input pictures? I thought steps were supposed to be relative to the number of input pictures

**Reply by TheLastBen** [reply]
not for the text_encoder, it only needs 350-600 steps for 20-60 pictures

**Reply by TetraOrb** [reply]
@TheLastBen end up with some interesting results. I used 57 pictures. Trained for 3000 steps and 350 steps for the text encoder. And now the model applies a style to everything in the picture, except faces. =)
![00221-765320212-art by tetraorb-](https://user-images.githubusercontent.com/115459677/206792973-0baea68a-aea8-4417-bb12-dd3b3209f3d9.png)
![00219-2038903917-art by tetraorb](https://user-images.githubusercontent.com/115459677/206792986-29cb1d37-39c3-4d2a-9fc1-575a2686c9ed.png)
Maybe more training is needed? Also, for earlier trainings I also used a batch of 100 and 300 images, can you please tell how the number of encoder training steps should be adjusted for those numbers?

**Reply by TheLastBen** [reply]
start with 2000 steps UNet and 350 steps text_enc, and make sure you use only one word for the images names a word that doesn't exist like "otiittre"

**Reply by TetraOrb** [reply]
Yes, I use one unexisting word. I'll try again, thank you! =)

**Reply by TetraOrb** [reply]
@TheLastBen tested the training today. Faces become stylized after 5000 steps. Really love the outcome! Thank you for al your work!

**Reply by dvelm** [reply]
What does it mean stylized? Does it mean that the face appears everytime in any prompt? Or it means that when the identifier word is written the outputs show the faces correctly?

**Reply by TetraOrb** [reply]
> What does it mean stylized? Does it mean that the face appears everytime in any prompt? Or it means that when the identifier word is written the outputs show the faces correctly?

I try ro teach SD to draw in my style. And on lower Unet steps it tends to make painted people with weird realistic faces like this: 
![00596-3375671143-portrait by ttrraa, hd, artstation trending, visual development, visdev](https://user-images.githubusercontent.com/115459677/206979207-7fc5926d-d1a4-48b1-9989-b93fecef7934.png)

But the more Unet I train, the better the result is for the faces. But I haven't tried training with reference images yet.
